### PR TITLE
Add some Princeton projects in the context of ROOT

### PR DESCRIPTION
**Pull request by vgvassilev**
Depends on #662

### CODE REVIEW

File: gsoc/2020/mentors.md [hunk 1]
```diff
@@ -17,3 +17,4 @@ layout: plain
 * Jim Pivarski [pivarski@princeton.edu](mailto:pivarski@princeton.edu) Princeton University
 * Alex Richards [a.richards@imperial.ac.uk](mailto:a.richards@imperial.ac.uk) Imperial College London
 * Mark Smith [mark.smith1@imperial.ac.uk](mailto:mark.smith1@imperial.ac.uk) Imperial College London
+* Vassil Vassilev [vvasilev@cern.ch](mailto:vvasilev@cern.ch) Princeton University
```
Comment (xvallspl) [inline gsoc/2020/mentors.md]: You don't want to add Alexander Penev here as well?

Comment (xvallspl) [reply gsoc/2020/mentors.md]: Or Bertrand!

File: _gsocproposals/2020/proposal_CLAD-NumericalDiff.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+## Mentors
+
+  * [Vassil Vassilev](mailto:vvasilev@cern.ch)
+  * [Alexander Penev](mailto:alexander_penev@yahoo.com)
+
+## Links
+
+  * [clad](https://github.com/vgvassilev/clad)
+  * [clad gsoc page](https://github.com/vgvassilev/clad/wiki/GSoC-2020)
+  * [clang](clang.llvm.org)
```
Comment (agheata) [inline _gsocproposals/2020/proposal_CLAD-NumericalDiff.md]: the link does not work, needs to be explicit

File: _gsocproposals/2020/proposal_CLAD-NumericalDiff.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+Intermediate C++; Understanding numerical differentiation; Understanding of Clang and its API.
+
+## Mentors
+
+  * [Vassil Vassilev](mailto:vvasilev@cern.ch)
+  * [Alexander Penev](mailto:alexander_penev@yahoo.com)
+
+## Links
+
+  * [clad](https://github.com/vgvassilev/clad)
```
Comment (agheata) [inline _gsocproposals/2020/proposal_CLAD-NumericalDiff.md]: I would be more verbose about what are the links: here "clad GitHub repository" ?

File: _gsocproposals/2020/proposal_CLAD-NumericalDiff.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+Currently, clad cannot differentiate declared-but-not-defined functions. In that case it issues an error. Instead, clad should fall back to its future numerical differentiation facilities.
+
+## Task ideas and expected results
+
+Implement numerical differentiation support in clad. It should be available through a dedicated interface (for example clad::num_differentiate). The new functionality should be connected to the forward mode automatic differentiation. If time permits, a prototype of configurable error estimation for the numerical differentiation should be implemented. The student should be prepared to write a progress report and present the results at the end of the summer.
```
Comment (agheata) [inline _gsocproposals/2020/proposal_CLAD-NumericalDiff.md]: Why don't you add: "More detailed information about the project and applicant guidelines are available in the [clad GSoC 2020] page." (with a direct link in the text. That would call in my opinion the interested students to click it and see your nice GSoC 2020 page...

File: gsoc/2020/mentors.md (hunk 1 above)
Comment (agheata) [reply gsoc/2020/mentors.md]: "and Bertrand" -  all project mentors have to be added here

File: _gsocproposals/2020/proposal_ROOTCxxModulesOnWindows.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+ROOT has several features which interact with libraries and require implicit header inclusion. This can be triggered by reading or writing data on disk, or user actions at the prompt. Often, the headers are immutable and reparsing is redundant. C++ Modules are designed to minimize the reparsing of the same header content by providing an efficient on-disk representation of C++ Code.
+
+In ROOT v6.20 C++ modules are default on UNIX and will be default on OSX in the next release. The next challenge is C++ Modules on Windows.
+
+## Task ideas and expected results
+
+The main task is to investigate the feasibility of the C++ modules technology on Windows. This can be done by trying to compile ROOT on windows with -Druntime_cxxmodules=On.
+
+The expected results is to enumerate a set Windows-specific issues and implement compatible to the UNIX baseline solutions. The student should be prepared to write a progress report and present the results at the end of the summer.
```
Comment (agheata) [inline _gsocproposals/2020/proposal_ROOTCxxModulesOnWindows.md]: a set of Windows-specific issues and implement solutions compatible to the UNIX baseline ones

File: _gsocproposals/2020/proposal_ROOTCxxModulesOnWindows.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+## Mentors
+
+  * [Vassil Vassilev](mailto:vvasilev@cern.ch)
+  * [Bertrand Bellenot](mailto:Bertrand.Bellenot@cern.ch)
+
+## Links
+
+  * [ROOT](http://root.cern)
+  * [clang](clang.llvm.org)
```
Comment (agheata) [inline _gsocproposals/2020/proposal_ROOTCxxModulesOnWindows.md]: link not valid, make it explicit

**Comment by vgvassilev** [comment]
All good points. I think all are addressed now.

**Pull request merged**